pymc_marketing/clv/models/basic.py

```diff
@@ -49,21 +43,80 @@ def fit(self, fit_method="mcmc", **kwargs):
         kwargs:
             Other keyword arguments passed to the underlying PyMC routines
         """
+
+        self.build_model()
+
         if fit_method == "mcmc":
-            res = self._fit_mcmc(**kwargs)
+            self._fit_mcmc(**kwargs)
         elif fit_method == "map":
-            res = self._fit_MAP(**kwargs)
+            self._fit_MAP(**kwargs)
         else:
             raise ValueError(
                 f"Fit method options are ['mcmc', 'map'], got: {fit_method}"
             )
-        self.fit_result = res
-        return res
+        self.idata.add_groups(fit_data=self.data.to_xarray())  # type: ignore
```

Comment:
Can you selectively suppress the warning that arviz issues about `fit_data` not being a default group?

CC @OriolAbril is there any thing we should pay attention (why is there a warning like that)?

Reply:
It's something that would need to be done on Arviz side, I've asked @OriolAbril about that before as it was messing with tests in pymc-experimental, he said that it wasn't anything to be worried about (back then)

Reply:
A PR to add a flag/boolear arg to add_groups and/or inferencedata init would also be welcome. It is generally useful to have that because using groups that are not used by any arviz function is quite uncommon, as opposed to making typos with the group names for example (the original reason for the warning IIRC). There is nothing wrong with using different group names though, the inferencedata schema is even explicit about that, so not being able to turn these off is anoying for cases like this.

Reply:
I've suppressed the warning with:
        with warnings.catch_warnings():
            warnings.filterwarnings(
                "ignore",
                category=UserWarning,
                message="The group fit_data is not defined in the InferenceData scheme",
            )
            self.idata.add_groups(fit_data=self.data.to_xarray())  # type: ignore
for now should be good enough as the users won't see the warning